modules/optflow/samples/gpc_evaluate.cpp

```diff
@@ -0,0 +1,163 @@
+#include "opencv2/core/ocl.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/optflow.hpp"
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+
+/* This tool finds correspondences between two images using Global Patch Collider
+ * and calculates error using provided ground truth flow.
+ *
+ * It will look for the file named "forest.dump" with a learned forest.
+ * You can obtain the "forest.dump" either by manually training it using another tool with *_train suffix
+ * or by downloading one of the files trained on some publicly available dataset from here:
+ *
+ * https://drive.google.com/open?id=0B7Hb8cfuzrIIZDFscXVYd0NBNFU
+ */
+
+using namespace cv;
+
+const String keys = "{help h ?     |           | print this message}"
+                    "{@image1      |<none>     | image1}"
+                    "{@image2      |<none>     | image2}"
+                    "{@groundtruth |<none>     | path to the .flo file}"
+                    "{@output      |           | output to a file instead of displaying, output image path}"
+                    "{f forest     |forest.dump| path to the forest.dump}";
+
+const int nTrees = 5;
+const bool useOpenCL = true;
+
+static double normL2( const Point2f &v ) { return sqrt( v.x * v.x + v.y * v.y ); }
```

Comment:
Btw, I've seen some pieces of code where people get around this by writing like
`sqrt(pt.dot(pt))`